Additional fix for missing pandas.Panel

enricodetoma opened the pull request:
I made an additional fix to manage the case of missing `Panel` in new `pandas` versions.
In fact I discovered that in few cases (e.g. `library.delete(symbol)`), `PandasPanelStore` static method `can_write_type` was still called.
With this commit, `PandasPanelStore` is not even registered as storage class if `Panel` is missing from `pandas`.

enricodetoma committed:
Additional fix for missing pandas.Panel

shashank88 commented:
@enricodetoma you need to rebase off the latest master after pinning pandas and numpy

enricodetoma commented:
@shashank88 I rebased off the latest master, but I'm not sure what you mean about pinning pandas and numpy.
This pull request completes my previous pull request #881 which was meant to allow to use Arctic with recent pandas versions, though without Panel support (but then without the need to pin it to a specific version, I guess).

enricodetoma commented:
Actually I'm already using Arctic successfully with pandas 1.2.3 with this change, without Panel support

crazy25000 commented:
> Actually I'm already using Arctic successfully with pandas 1.2.3 with this change, without Panel support

Same here, using it with latest Pandas. I don't use panels explicitly, just using tick store. Chunk store too slow to use.

enricodetoma committed:
With the fix for missing pandas.Panel it is no longer needed to force old numpy/pandas versions

enricodetoma committed:
Merge branch 'man-group:master' into master

enricodetoma committed:
Merge remote-tracking branch 'man-group/master'

enricodetoma committed:
Merge branch 'man-group:master' into master

enricodetoma committed:
Merge remote-tracking branch 'man-group/master'

# Conflicts:
#	setup.py

enricodetoma committed:
Merge branch 'master' of github.com:enricodetoma/arctic

enricodetoma committed:
Merge remote-tracking branch 'man-group/master'

samuelrohr commented:
Any update on this review?

enricodetoma committed:
Merge tag 'v1.80.5'

tagging v1.80.5

enricodetoma committed:
Removed dependencies on old numpy and pandas packages

dunckerr reviewed (commented):
Hi Enrico - we haven’t removed Panel support because we are still supporting old versions of Pandas. Am I correct that you’re forking arctic repo and removing Panel support?  We are always interested to hear about any improvements to the arctic codebase.
Thanks
Duncan

enricodetoma commented:
Hi Duncan, I'm simply skipping importing Panel if it doesn't exist:

    try:
        from pandas import Panel
        register_versioned_storage(PandasPanelStore)
    except ImportError:
        pass

enricodetoma committed:
pymongo <= 3.13.0

enricodetoma committed:
Merge remote-tracking branch 'man-group/master'

# Conflicts:
#	arctic/__init__.py
#	setup.py

enricodetoma committed:
pymongo <= 3.13.0

enricodetoma committed:
pandas 2 and numpy 2

enricodetoma committed:
pymongo <4

enricodetoma committed:
numpy<2

enricodetoma committed:
-

enricodetoma committed:
Fix to prevent using an old version of numpy<2 for setup

enricodetoma committed:
Fix pandas 2.2+/3.x compat by falling back to pickle_compat.loads

pandas dropped the file-object-based pickle_compat.load in favour of
a bytes-based loads, breaking import of arctic.store.version_store.
